core/src/test/scala/unit/kafka/admin/ConsumerGroupServiceTest.scala

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.admin
+
+import java.util
+import java.util.{Collections, Optional}
+
+import kafka.admin.ConsumerGroupCommand.{ConsumerGroupCommandOptions, ConsumerGroupService}
+import org.apache.kafka.clients.admin._
+import org.apache.kafka.clients.consumer.{OffsetAndMetadata, RangeAssignor}
+import org.apache.kafka.common.{ConsumerGroupState, KafkaFuture, Node, TopicPartition, TopicPartitionInfo}
+import org.easymock.EasyMock
+import org.easymock.EasyMock._
+import org.junit.jupiter.api.Assertions.{assertEquals, assertTrue}
+import org.junit.jupiter.api.Test
+
+import scala.jdk.CollectionConverters._
+
+class ConsumerGroupServiceTest {
+
+  private val group = "testGroup"
+  private val topics = (0 until 5).map { i => s"testTopic$i" }
+  private val numPartitions = 10
+  private val topicPartitions = topics.flatMap { topic => (0 until numPartitions).map { i => new TopicPartition(topic, i) }}
+  private val admin: Admin = mock(classOf[Admin])
+
+  @Test
+  def testAdminRequestsForDescribeOffsets(): Unit = {
+    val args = Array("--bootstrap-server", "localhost:9092", "--group", group, "--describe", "--offsets")
+    val groupService = consumerGroupService(args)
+
+    expect(admin.describeConsumerGroups(EasyMock.eq(Collections.singletonList(group)), anyObject()))
+      .andReturn(describeGroupsResult(ConsumerGroupState.STABLE))
+      .once()
+    expect(admin.listConsumerGroupOffsets(EasyMock.eq(group), anyObject()))
+      .andReturn(listGroupOffsetsResult)
+      .once()
+    expect(admin.listOffsets(anyObject(), anyObject()))
+      .andAnswer(() => listOffsetsResult(getCurrentArguments.apply(0).asInstanceOf[util.Map[TopicPartition, OffsetSpec]]))
+      .once()
+    replay(admin)
+
+    val (state, assignments) = groupService.collectGroupOffsets(group)
+    assertEquals(Some("Stable"), state)
+    assertTrue(assignments.nonEmpty)
+    assertEquals(topicPartitions.size, assignments.get.size)
+  }
+
+  @Test
+  def testAdminRequestsForResetOffsets(): Unit = {
+    val args = Array("--bootstrap-server", "localhost:9092", "--group", group, "--reset-offsets", "--to-latest")
+    val topicsWithoutPartitionsSpecified = topics.tail
+    val topicArgs = Seq("--topic", s"${topics.head}:${(0 until numPartitions).mkString(",")}") ++
+      topicsWithoutPartitionsSpecified.flatMap { topic => Seq("--topic", topic) }
+    val groupService = consumerGroupService(args ++ topicArgs.toArray)
+
+    expect(admin.describeConsumerGroups(EasyMock.eq(Collections.singletonList(group)), anyObject()))
+      .andReturn(describeGroupsResult(ConsumerGroupState.DEAD))
+      .once()
+    expect(admin.describeTopics(EasyMock.eq(topicsWithoutPartitionsSpecified.asJava), anyObject()))
+      .andReturn(describeTopicsResult)
+      .once()
+    expect(admin.listOffsets(anyObject(), anyObject()))
```

Comment:
Would it be possible to specify the expected input for the first argument here? I see that you have assertions about it in `listOffsetsResult` but it might be better to be explicit here directly if we can. I was not expecting `listOffsetsResult` to do any assertions at first.

Reply:
Couldn't compare with `eq` since OffsetSpec used as value in the Map doesn't have `equals`. Added an argument matcher.